modules/transcription-service-microsoft-azure/src/main/java/org/opencastproject/transcription/microsoftazure/MicrosoftAzureTranscriptionService.java

```diff
@@ -499,37 +487,37 @@ void recognizeContinuous(SpeechRecognizer speechRecognizer, String jobId, String
     final int[] sequenceNumber = new int[] { 0 };
 
     speechRecognizer.sessionStarted.addEventListener((s, e) -> {
+      logger.trace("Transcription job {} started on media package {}.", jobId, mpId);
       try {
         database.storeJobControl(mpId, track.getIdentifier(), jobId, TranscriptionJobControl.Status.InProgress.name(),
                 track.getDuration() == null ? 0 : track.getDuration().longValue(), null, PROVIDER);
 
-        if (!useSubRipTextCaptionFormat) {
-          try {
-            writeToTmpFile(String.format("WEBVTT%s%s", System.lineSeparator(), System.lineSeparator()), jobId);
-          } catch (IOException ioException) {
-            logger.error(String.format("Could not write header to tmp file"));
-          }
-        }
+        createTranscriptFile(jobId);
       } catch (TranscriptionDatabaseException ex) {
-        errorCallback("Could not store job: " + ex, speechRecognizer, jobId, mpId);
+        errorCallback("Could not store transcript job: " + ex, speechRecognizer, jobId, mpId);
+      }  catch (IOException ex) {
+        errorCallback("Unable to create transcript file: " + ex, speechRecognizer, jobId, mpId);
       }
-      return;
     });
 
     // Fired when an utterance is completely recognized
     speechRecognizer.recognized.addEventListener((s, e) -> {
       if (ResultReason.RecognizedSpeech == e.getResult().getReason() && e.getResult().getText().length() > 0) {
         sequenceNumber[0]++;
         String text = captionFromSpeechRecognitionResult(sequenceNumber[0], e.getResult());
-
         try {
-          writeToTmpFile(text, jobId);
-        } catch (IOException ioException) {
-          logger.error(String.format("Could not write to tmp file: %s", text));
+          writeTranscriptToFile(text, jobId);
```

Comment:
Why are we creating the file, then recognizing, and only then writing to the file?  Why not just create the file when it's time to write?